src/wagtail_ai/static_src/field_panel/main.ts

```diff
@@ -57,15 +166,21 @@ class PromptController extends Controller<HTMLButtonElement> {
   async prompt(
     event?: CustomEvent<PromptOptions> & { params?: PromptOptions },
   ) {
-    const {
-      promptId,
-      method = PromptMethod.APPEND,
-      useContent = false,
-    } = {
+    const { promptId = this.activePromptIdValue } = {
       ...event?.detail,
       ...event?.params,
     };
-    const icon = this.element.querySelector('svg use');
+    this.stateValue = FieldPanelState.LOADING;
+    this.activePromptIdValue = promptId;
+    // The setter above only sets the data attribute, and Stimulus runs the
+    // callback asynchronously when the MutationObserver notices the change.
+    // Call the callback manually to ensure `this.activePrompt` is set.
+    this.activePromptIdValueChanged();
```

Comment:
I didn't know this about Stimulus. I thought setting a value directly in JS would immediately call the value changed callback before the next line, but apparently not. Or maybe I missed something...